src/compilation/compiler.rs

```diff
@@ -462,9 +462,17 @@ impl Compiler {
     }
 
     /// Compiles an integer literal expression.
+    /// 
+    /// This function converts a `TypedIntLiteral` into an LLVM constant integer value.
+    /// If `int_literal.negative` is true, the value is negated.
     unsafe fn compile_int_literal(&self, int_literal: &TypedIntLiteral) -> LLVMValueRef {
         let int_type = self.to_llvm_type(&Type::Int(int_literal.int_type));
-        let value_cstr = CString::new(int_literal.int_value.as_str()).unwrap();
+        let value_str = if int_literal.negative {
+            format!("-{}", int_literal.int_value)
+        } else {
+            int_literal.int_value.clone()
+        };
```

Comment:
Just for consistency with other parts of the code -- I'm good either way!

```suggestion
        let value_str = match int_literal.negative {
            true => format!("-{}", int_literal.int_value),
            false => int_literal.int_value.clone(),
        };
```